test/unittests/test_pla_parser.cpp

```diff
@@ -0,0 +1,80 @@
+#include "core/truthTable/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace syrec;
+
+class PlaParserTest: public testing::TestWithParam<std::string> {
+protected:
+    std::string       test_circuits_dir = "./circuits/";
+    syrec::TruthTable testPla;
+};
+
+TEST_F(PlaParserTest, andTest) {
+    std::string circAnd = test_circuits_dir + "and.pla";
+
+    EXPECT_EQ(testPla.nInputs(), 0U);
+    EXPECT_EQ(testPla.nOutputs(), 0U);
+    EXPECT_EQ(testPla.ioCube().size(), 0U);
+
+    EXPECT_TRUE(read_pla(testPla, circAnd));
+
+    EXPECT_EQ(testPla.nInputs(), 3U);
+    EXPECT_EQ(testPla.nOutputs(), 3U);
+    EXPECT_EQ(testPla.ioCube().size(), 4U);
```

Comment:
These checks will have to be adapted based on the changes requested above. Merely reading a PLA file should not involve any embedding or encoding.

If you want to test that behaviour as well, then write separate and explicit tests for that.